src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts

```diff
@@ -172,10 +174,27 @@ export class UBSOrderDetailsComponent implements OnInit, OnDestroy {
     }
   }
 
-  clearOrderValues(): void {
-    this.additionalOrders.controls.forEach((element) => {
-      element.setValue('');
+  public ecoStoreValidation() {
+    let counter = 0;
+    this.additionalOrders.controls.forEach((controller) => {
+      if (controller.valid && controller.dirty && controller.value !== '') {
+        counter++;
+      }
     });
+
+    if (counter === this.additionalOrders.controls.length) {
+      this.displayOrderBtn = true;
+    } else {
+      this.displayOrderBtn = false;
+    }
+  }
+
+  clearOrderValues(): void {
+    this.additionalOrders.controls[0].setValue('');
+    if (this.additionalOrders.controls.length > 1) {
+      this.additionalOrders.controls.splice(1);
+    }
+    this.ecoStoreValidation();
   }
```

Comment:
it's a bit wrong logic here 
it should always show input 
but if the user start using it - checkbox yes should be selected ( automatically  )

Reply:
![image](https://user-images.githubusercontent.com/43093994/120103454-53ac0f80-c158-11eb-892c-92ea91fe763f.png)